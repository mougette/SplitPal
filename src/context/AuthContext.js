import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
