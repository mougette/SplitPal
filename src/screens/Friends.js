import React, { useState, useContext, useEffect } from 'react';
import {View, Text, StyleSheet, Button, TextInput, FlatList, SafeAreaView, TouchableOpacity, ImageBackground, Alert} from 'react-native';
import Entry from "../components/EntryComponent";
import SearchAndAdd from "../components/SearchAndAddComponent";
import {Get} from '../components/RestGet';
import Patch from '../components/RestPatch';
import { Context as AuthContext } from "../context/AuthContext";


const Friends = ({navigation}) => {
const { state, setState2 } = useContext(AuthContext);
const [DATA, setDATA] = useState("");
const [DATA2, setDATA2] = useState("");
useEffect(() => {
const unsubscribe = navigation.addListener('focus', () => {
Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/friend","?user="+state.email)
   .then(response => setDATA(JSON.parse(response)));
Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/friend-request","?user="+state.email)
   .then(response => setDATA2(JSON.parse(response)));
   });
     return unsubscribe
},[navigation]);
  const renderItem = ( {item,index} ) => {
  if(item.Balance == "Accept"){
  return (
          <Entry
              image='https://reactnative.dev/img/tiny_logo.png'
              name={item.FirstName+item.LastName}
              balance={item.Balance}
              onPress = { () => {
              Alert.alert(
                  "Accept friend request?",
                  "",
              [
              {
              text: "Yes",
              onPress: () =>{
              Patch("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/friend-request",
              JSON.stringify({
                    userEmail: state.email,
                    friendEmail: item.Email,
                    accepted : "1",
                    }));
              setTimeout(() => {
                  Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/friend","?user="+state.email)
                  .then(response => setDATA(JSON.parse(response)));
                  console.log(DATA)
                  Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_2/friend-request","?user="+state.email)
                  .then(response => setDATA2(JSON.parse(response)));
                  console.log(DATA2)
                  }, 2000);
                    },
              },
              {
                            text: "No",
                            onPress: () =>{
                            Patch("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/friend-request",
                            JSON.stringify({
                                  userEmail: state.email,
                                  friendEmail: item.Email,
                                  accepted : "0",
                                  }));

                            setTimeout(() => {
                            Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/friend","?user="+state.email)
                               .then(response => setDATA(JSON.parse(response)));
                             console.log(DATA)
                            Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/friend-request","?user="+state.email)
                               .then(response => setDATA2(JSON.parse(response)));
                               console.log(DATA2)
                                  }, 2000);
                                  },
              },
              ]
              );
              }
              }
          ></Entry>
      );
  }
  else if(item.Balance == "Split"){
      return (
          <Entry
              image='https://reactnative.dev/img/tiny_logo.png'
              name={item.FirstName+item.LastName}
              balance={item.Balance}
              onPress = { () => navigation.navigate("Split", [{item}, {state}, "NULL"]) }
          ></Entry>
      );
      }
  else{
    return (
        <Entry
            image='https://reactnative.dev/img/tiny_logo.png'
            name={item.FirstName+item.LastName}
            balance={item.Balance}
            onPress = { () => console.log("Yo!") }
        ></Entry>
    );
    }
  };

  return (
    <View style={styles.master}>
      <SearchAndAdd search={"Search Friends"} button={"Add Friend"} onPress = { () => navigation.navigate("Add Friends") } />
        <SafeAreaView style={styles.containerTop}>
          <FlatList
              data={DATA}
              renderItem={renderItem}
              keyExtractor = {(item, index) => index.toString()}
              extraData={DATA}
          />
        </SafeAreaView>
        <Text style={styles.header}>Pending Requests</Text>
        <SafeAreaView style={styles.containerBot}>
          <FlatList
              data={DATA2}
              renderItem={renderItem}
              keyExtractor = {(item, index) => index.toString()}
              extraData={DATA2}
          />
        </SafeAreaView>
    </View>
  );
};

const styles = StyleSheet.create({
  master: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#87a1b6'
  },
  header: {
    fontSize: 32,
  },
  blueTextButton: {
    fontSize: 20,
    borderWidth: 1,
    borderRadius: 7,
    overflow: "hidden",
    padding: 10,
    marginLeft: 10,
    color: '#2196F3',
    backgroundColor: 'white',
  },
  link: {
    flexDirection: 'row',
    justifyContent: 'center',
    paddingTop: 20,
  },
  input: {
    width: '60%',
    borderWidth: 1,
    borderRadius: 7,
    padding: 10,
    backgroundColor: 'white',
  },
  containerTop: {
    flex: 2,
    marginTop: 15,
  },
  containerBot: {
    flex: 1,
    marginTop: 0,
  },
  item: {
    padding: 20,
    marginVertical: 8,
    paddingHorizontal: '35%',
    borderRadius: 5,
  },
  image: {
    flex: 1,
        justifyContent: "center"
  },
});

export default Friends;
