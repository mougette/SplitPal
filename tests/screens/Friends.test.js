import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import Friends from '../../src/screens/Friends'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert } from 'react-native';


global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("[{\"FirstName\": \"2\", \"LastName\": \"2\", \"Email\": \"abc@e.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"fake\", \"LastName\": \"1\", \"Email\": \"fake1@gmail.com\", \"Balance\": \"Pending\"}, {\"FirstName\": \"fake\", \"LastName\": \"5\", \"Email\": \"fake5@gmail.com\", \"Balance\": \"Accept\"}, {\"FirstName\": \"hi\", \"LastName\": \"there\", \"Email\": \"newEmail@test.com\", \"Balance\": \"Split\"}, {\"FirstName\": \"Sam\", \"LastName\": \"Hogenson\", \"Email\": \"shogenson@wisc.edu\", \"Balance\": \"Split\"}, {\"FirstName\": \"test\", \"LastName\": \"today\", \"Email\": \"testing@test.com\", \"Balance\": \"Split\"}]"),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Friends', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Click Split', async () => {
                const {getAllByText} = render(
                <AuthProvider>
                <NavigationContainer>
                <Stack.Navigator>
                <Stack.Screen name="Friends" component={Friends} />
                </Stack.Navigator>
                </NavigationContainer>
                </AuthProvider>);
                await waitFor(() => {
                fireEvent.press(getAllByText("Split")[0]);
                expect(1).toBe(1);
                });
                })
    it('Click Accept', async () => {
    jest.useFakeTimers()
    Alert.alert = jest.fn();
                    const {getAllByText} = render(
                    <AuthProvider>
                    <NavigationContainer>
                    <Stack.Navigator>
                    <Stack.Screen name="Friends" component={Friends} />
                    </Stack.Navigator>
                    </NavigationContainer>
                    </AuthProvider>);
                    await waitFor(() => {
                    fireEvent.press(getAllByText("Accept")[0]);
                    Alert.alert.mock.calls[0][2][0].onPress()
                    jest.advanceTimersByTime(3000)
                    Alert.alert.mock.calls[0][2][1].onPress()
                    jest.advanceTimersByTime(3000)
                    expect(Alert.alert.mock.calls.length).toBe(1);
                    });
                    })
    it('Click Add Friend', async () => {
                const {getByText} = render(
                <AuthProvider>
                <NavigationContainer>
                <Stack.Navigator>
                <Stack.Screen name="Friends" component={Friends} />
                </Stack.Navigator>
                </NavigationContainer>
                </AuthProvider>);
                await waitFor(() => {
                fireEvent.press(getByText("Add Friend"));
                expect(1).toBe(1);
                });
                })


});