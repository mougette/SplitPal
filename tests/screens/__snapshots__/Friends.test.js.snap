// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends Click Add Friend 1`] = `undefined`;
